Desktop_Interface/isobuffer.cpp

```diff
@@ -51,49 +57,58 @@ void isoBuffer::insertIntoBuffer(short item)
 
 short isoBuffer::bufferAt(int idx) const
 {
-	return m_buffer[m_back - idx];
+	// NOTE: this is only correct if idx < m_insertedCount
+	return m_buffer[m_back + (m_bufferEnd+1) - idx];
```

Comment:
Whoa, good catch.  That would have caused hell for high values of idx otherwise.